packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientOptionsProvider.cs

```diff
@@ -6,25 +6,161 @@
+        }
+
+        protected override ConstructorProvider[] BuildConstructors()
+        {
+            if (_serviceVersionDefinition == null || _latestVersionField == null)
+                return [];
+
+            var versionParam = new ParameterProvider(
+                "version",
+                $"The service version",
+                _serviceVersionDefinition.Type,
+                defaultValue: _latestVersionField);
+            var serviceVersionsCount = _serviceVersionDefinition.Members.Count;
+            List<SwitchCaseExpression> switchCases = new(serviceVersionsCount + 1);
+
+            for (int i = 0; i < serviceVersionsCount; i++)
+            {
+                var serviceVersionMember = _serviceVersionDefinition.Members[i];
+                // ServiceVersion.Version => "version"
+                switchCases.Add(new(
+                    new MemberExpression(_serviceVersionDefinition.Type, serviceVersionMember.Name),
+                    new LiteralExpression(serviceVersionMember.Value)));
+            }
+
+            switchCases.Add(SwitchCaseExpression.Default(ThrowExpression(New.NotSupportedException(ValueExpression.Empty))));
+
+            var constructor = new ConstructorProvider(
+                new ConstructorSignature(Type, $"Initializes a new instance of {_clientOptionsName}.", MethodSignatureModifiers.Public, [versionParam]),
+                _versionProperty!.Assign(new SwitchExpression(versionParam, [.. switchCases])).Terminate(),
+                this);
+            return [constructor];
+        }
+
+        protected override PropertyProvider[] BuildProperties()
+        {
+            List<PropertyProvider> properties = _versionProperty != null ? [_versionProperty] : [];
+
+            foreach (var f in _clientProvider.Fields)
+            {
+                if (f != _clientProvider.ApiKeyAuthField
+                    && f != _clientProvider.EndpointField
+                    && !f.Modifiers.HasFlag(FieldModifiers.Const))
+                {
+                    properties.Add(new(
+                        f.Description,
+                        MethodSignatureModifiers.Public,
+                        f.Type.PropertyInitializationType,
+                        f.Name.ToCleanName(),
+                        new AutoPropertyBody(true),
+                        backingField: f));
+                }
+            }
+
+            return [.. properties];
+        }
+
+        private static ApiVersion[] ParseApiVersions(IReadOnlyList<string> inputApiVersions)
+        {
+            if (inputApiVersions == null)
+            {
+                return [];
+            }
+
+            var count = inputApiVersions.Count;
+            var parsedVersions = new ApiVersion[count];
+
+            for (int i = 0; i < count; i++)
+            {
+                var normalizedVersion = NormalizeVersion(inputApiVersions[i]);
+                var description = $"Service version \"{inputApiVersions[i]}\"";
```

Comment:
So if we change the type of description to `FormattableString` and pass this type around, we could simply do this:
```suggestion
                var description = $"Service version {inputApiVersions[i]:L}";
```
In this way CodeWriter should handle the literal properly and we do not need to worry about it